### PR TITLE
Feature: Add `{{ns}}` helper and tests

### PR DESCRIPTION
**Pull request by erikjung**
Looking into the feasibility of using PostHTML to automatically prefix classes in HTML with `drizzle-` made me realize a challenge I hadn't considered. 

Because it would need to be done after the Handlebars rendering, there would be no way to differentiate Drizzle UI elements from patterns:

```html
<!-- <drizzle> -->
<div class="u-pad u-cf u-posRelative ">

  <!-- <pattern> -->
  <button class="Button Button--primary">
    Primary Button
  </button>
  <!-- </pattern> -->

</div>
<!-- </drizzle> -->
```

All of the classes (even the pattern ones) would end up with the prefix. The PostHTML [plugin](https://github.com/stevenbenisek/posthtml-prefix-class) provides a way to ignore certain classes, but it doesn't seem applicable because the class naming schemes of both the patterns and the Drizzle UI are typically identical.

So, I thought another solution might be to rely on a Handlebars helper for the prefixing.

It works like this:

```hbs
{{ns "some space-separated words" prefix="ABC-"}}
{{!  ABC-some ABC-space-separated ABC-words  }}
```

```hbs
<div class="{{ns 'u-pad u-cf u-posRelative' prefix='drizzle-'}}"> 
  <button class="Button Button--primary">
    Primary Button
  </button>
</div>

{{!
<div class="drizzle-u-pad drizzle-u-cf drizzle-u-posRelative">
  <button class="Button Button--primary">
    Primary Button
  </button>
</div>
}}
```

The `drizzle-` prefix is set as a default so it isn't needed each time.

/CC @tylersticka @saralohr @nicolemors

### CODE REVIEW

**Comment by tylersticka** [comment]
What about in cases where the class is passed via Frontmatter? [Example](https://github.com/cloudfour/drizzle/blob/b9c0592a89a371f4b9748c87034c6f9d985638bb/src/patterns/typography/headings.hbs#L9)

The issue there is that we don't really want to prevent usage of non-namespaced classes. So maybe that's a lost cause. Just figured I'd mention it, since day-to-day I probably _use_ the namespaced classes via the Frontmatter more than in the templates themselves.

👍  for non-user-impacting dev convenience, though!

**Comment by erikjung** [comment]
@tylersticka 

> What about in cases where the class is passed via Frontmatter?

Good catch. You would just use another `{{ns}}` for that value:

```yaml
---
previewClass: u-rhythm
---
```

```hbs
<div class="{{ns 'u-pad'}} {{ns previewClass}}"></div>

{{!
<div class="drizzle-u-pad drizzle-u-rhythm"></div>
}}
```

**Comment by tylersticka** [comment]
Except...

> The issue there is that we don't really want to prevent usage of non-namespaced classes.

In the Cloud Four Redesign project, sometimes we want to use a `drizzle-*` class, sometimes we want to use a toolkit class (like `.Sky`).

**Comment by erikjung** [comment]
@tylersticka 

Ahhhhh. Now your comment makes more sense. I didn't know we ever added non-Drizzle classes to the `previewClass` property, because they're output on a "Drizzle element".

Looking at https://github.com/cloudfour/cloudfour.com-patterns/search?utf8=✓&q=previewClass

I would recommend that we don't use toolkit components or utilities in this way. If a utility is needed, we can add a new one to the Drizzle CSS (even if just in a specific project's `assets/drizzle/styles`) and if a _component_ is needed (which seems pretty rare), I think we should just include as a wrapper in the pattern source.

**Comment by tylersticka** [comment]
> I would recommend that we don't use toolkit components or utilities in this way. If a utility is needed, we can add a new one to the Drizzle CSS (even if just in a specific project's assets/drizzle/styles) and if a component is needed (which seems pretty rare), I think we should just include as a wrapper in the pattern source.

If I'm reading your suggestion correctly, it sounds like I would either have to:

1. Wrap the pattern body with `<div class="Sky"></div>` even though that component isn't really a requirement of what I'm demonstrating.
2. Make a new `drizzle-` class that does exactly what `Sky` does and maintain them separately.

If that's the case, I kinda think we should keep namespacing _out_ of this attribute. If I had to choose between _only_ Drizzle classes or _only_ toolkit classes for `previewClass`, I would easily choose `toolkit` classes. I can never imagine a scenario where `drizzle-u-rhythm` will be more appropriate than `u-spaceItems` (as an example) for showcasing how a pattern performs will look within the toolkit.

TL;DR: The preview element is for previewing stuff from the toolkit. It sounds backwards to say "you can only use drizzle classes to augment the preview of toolkit patterns." I feel bad for suggesting it.

**Comment by erikjung** [comment]
@tylersticka 

> If that's the case, I kinda think we should keep namespacing out of this attribute. If I had to choose between only Drizzle classes or only toolkit classes for previewClass, I would easily choose toolkit classes.

I'm okay with that, but we should make it consistent. Definitely something to document (https://github.com/cloudfour/drizzle/issues/31). 

Should we then update [this area](https://github.com/cloudfour/drizzle/blob/master/src/templates/drizzle/item.hbs#L34-L36) to have the output wrapped so that only _toolkit_ (non-prefixed) classes end up on the preview container?

```hbs
<div class="drizzle-u-pad drizzle-u-cf drizzle-u-posRelative">
  <div class="{{data.previewClass}}">
    {{{pattern id @root}}}
  </div>
</div>
```

**Comment by tylersticka** [comment]
I think where I'm struggling is that I don't mind the way it works _now_. 😆 

Could we just pretend I never said anything and leave `previewClass` as-is? 😬

**Comment by erikjung** [comment]
> Could we just pretend I never said anything and leave previewClass as-is? 😬

We can leave the HTML alone, but I think the intention for that `previewClass` should still be documented as a place for additional *tookit* classes to go (because Drizzle ones won't work due to the missing prefix, but we can omit that detail).

**Comment by tylersticka** [comment]
> We can leave the HTML alone, but I think the intention for that previewClass should still be documented as a place for additional tookit classes to go (because Drizzle ones won't work due to the missing prefix, but we can omit that detail).

Agreed. 👍

**Comment by mrgerardorodriguez** [comment]
👍
